test/common/websocket/codec_test.cc

```diff
@@ -0,0 +1,824 @@
+#include <array>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/websocket/codec.h"
+
+#include "test/common/buffer/utility.h"
+#include "test/test_common/printers.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace WebSocket {
+namespace {
+
+// A single-frame unmasked text message
+// 0x81 0x05 0x48 0x65 0x6c 0x6c 0x6f (contains "Hello")
+//                   H        e        l        l        o
+// 10000001 00000101 01001000 01100101 01101100 01101100 01101111
+// fin - 1
+// opcode - 0x1 - text data
+// mask bit - 0
+// length - 5 bytes 0000101
+TEST(WebSocketCodecTest, decodeSingleUnmaskedFrame) {
+  Buffer::OwnedImpl buffer;
+  Buffer::addSeq(buffer, {0x81, 0x05, 0x48, 0x65, 0x6c, 0x6c, 0x6f});
+
+  Decoder decoder;
+  absl::optional<std::vector<Frame>> frames = decoder.decode(buffer);
+
+  ASSERT_TRUE(frames.has_value());
```

Comment:
Let's also assert that `frames` has the right number of frames before we attempt to access any of the frames:
```
ASSERT_EQ(1, frames->size());
```
We should do this elsewhere in the file. (And we can remove the corresponding EXPECT_EQ on line 35, and elsewhere)